crates/client/mempool/src/block_production.rs

```diff
@@ -250,8 +251,11 @@ impl BlockProductionTask {
                     self.block.inner.transactions.push(converted_tx.transaction);
                 }
                 Err(err) => {
-                    // TODO: revert handling
-                    log::error!("Unsuccessful execution of transaction {:?}: {err:#}", mempool_tx.tx_hash());
+                    // These are the transactions that have errored but we can't revert them. It can be because of an internal server error, but
+                    // errors during the execution of Declare and DeployAccount also appear here as they cannot be reverted.
+                    // We reject them.
+                    // Note that this is a big DoS vector.
```

Comment:
this is how starknet is designed and we'll have to talk about it in full node discussions -- we can't fix that on our end and it reaaaally changes a lot of things

I believe the whole sierra to casm compiler would need to be rewritten entirely in cairo. so that class compilation is factored in the fees and the fees are verifiable in SNOS.